src/brasil/gov/portal/profiles/default/metadata.xml

```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0"?>
 <metadata>
-  <version>10803</version>
+  <version>10803r01</version>
```

Comment:
alguma razão para fazer ainda mais maluco o número da versão do profile? :)

prefiro deixar simplesmente `10804`.

Reply:
Rs, achei que o que estava no commit dava pra entender. É pra evitar conflitos quando for atualizar para a 2.x É o seguinte: estou na branch 1.x. Uma hora ela deixará de ser usada para atualizar para a master (2.0).

Imagine que eu criei o upgradeStep 10400 para essa branch 1.x, e depois o 10500 para atender alguma outra peculiaridade para esse branch porque o 2.x ainda não finalizou.

Teríamos o cenário: portal que estava na versão 1.5;1, no profile 10500, agora atualiza para o portal 2.x, que tem profiles com o mesmo número e implementações diferentes. Pior: estarei no 10500 e vai aparecer um upgradeStep 10400 para aplicar.

A solução menos dolorosa que encontramos pra evitar problemas em portais em produção, hoje, quando precisarem atualizar para a 2.x, foi essa.

Reply:
Esse número não é totalmente maluco, as lógicas distutils.version.LooseVersion
e packaging.version.parse conseguem dar um parse tranquilamente como pode ser visto em https://github.com/plonegovbr/brasil.gov.portal/pull/471/files#diff-18ab565e2ab82ef78f8bb92e8593518eR690.

Resumo: eu não queria usar esse formato, mas foi a forma que encontramos de causar menos problemas em ambientes legados.

Reply:
ah, tá... ich, não pensamos nisso, né? não seria melhor arrumar o profile e os upgrade steps no branch 2.0?

Reply:
atualmente a versão do profile no branch 2.0 é `10806`; eu sinceramente prefiro arrumar isso lá deixando como `20000` ou algo assim que complicar ainda mais as coisas.

o estado de 2.0 é alpha, então isso facilita todo; podemos indicar claramente que os releases anteriores da 2.0 não são suportados e não devem ser usados.

nós podemos ajudar em isso, só precisamos combinar aqui o que deve ser feito e arrumar lá.

Reply:
> o estado de 2.0 é alpha, então isso facilita todo; podemos indicar claramente que os releases anteriores da 2.0 não são suportados e não devem ser usados.

Você está falando dos releases 2.0ax, certo?

Reply:
certo, o que podemos fazer é no próximo release do 2.0 acertar a versão do profile e concentrar todos os upgrade step anteriores em um só.

precisaremos colocar também uma nota no changelog que não suportamos upgrades de uma versão menor a essa ai.

CC @rodfersou @agnogueira

Reply:
Qual número você vai escolher? Preciso disso também no brasil.gov.agenda. Favor me informar para que eu prossiga com a mudança do 10803r1 para 10804, e no brasil.gov.agenda do do 4003r01 para o 4004.

Então está confirmado que vocês alterarão isso no 2.0? Recomendo abrir um issue no brasil.gov.portal e no brasil.gov.agenda pra ficar documentado, pois depois iremos lançar um release 1.5.x com essas mudanças nos upgradeSteps e não terá mais como voltar, rs.

Reply:
Estamos atuando em https://github.com/plonegovbr/brasil.gov.agenda/tree/issue-82-erro-portlet-calendario-idgserpro, e no commit https://github.com/plonegovbr/brasil.gov.agenda/commit/453d40e7b807e92465c49e63d4457e54a5108615 temos a mesma situação de nomear um upgradeStep como r01. Me confirme quais versões vocês irão utilizar nas branches 2.x desses dois pacotes para que eu já possa corrigir. Preciso fazer um PR disso tudo com as correções hoje ainda.

Enfim, obrigado!

Reply:
@hvelarde Vou partir da premissa então que vocês irão partir de no mínimo 10900 no 2.x, ok? Aí farei 10804, e se for necessário outro upgradeStep no 1.5.x, será 10805 (teremos opção de quase 100 upgradeSteps antes de dar conflito, rs). Preciso fazer o rebase.

No caso do brasil.gov.agenda, vou partir da premissa de que vocês irão partir de pelo menos 4100 (ou 5000), e farei 4004.

Reply:
Só pra efeito de curiosidade, Plone teve um problema "parecido" com o fato de conflito em números dos profiles que comentamos em https://github.com/plonegovbr/brasil.gov.portal/pull/471#discussion_r187348711. Veja em https://github.com/plone/Products.CMFPlone/issues/2377